#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("bin_size", int_t, 0, "bin size of histogram", 100, 1, 256)
gen.add("queue_size", int_t, 0, "queue size of message", 100, 1, 1000)
gen.add("distance_threshold", double_t, 0, "filtering threshold", 0.6, 0.0, 1.0)
gen.add("flip_threshold", bool_t, 0, "flip threshold", False)

compare_policy_enum = gen.enum([gen.const("CORRELATION", int_t, 0, "use correlation"),
                                gen.const("BHATTACHARYYA", int_t, 1, "use bhattacharyya distance"),
                                gen.const("INTERSECTION", int_t, 2, "use histogram intersection"),
                                gen.const("CHISQUARE", int_t, 3, "use chi square function"),
                                gen.const("KL_DIVERGENCE", int_t, 4, "use KL divergence"),],
                               "histogram compare policy")
gen.add("compare_policy", int_t, 0, "compare policy", 0, 0, 4,
        edit_method=compare_policy_enum)

exit(gen.generate(PACKAGE, PACKAGE, "ColorHistogramFilter"))
